src/gt4py/next/program_processors/runners/dace_fieldview/gtir_dataflow.py

```diff
@@ -739,66 +733,149 @@ def _visit_map(self, node: gtir.FunCall) -> ValueExpr:
-            ndims = len(input_desc.shape) - 1
-            # the axis to be reduced is always the last one, because `reduce` is supposed
-            # to operate on `ListType`
-            assert set(input_expr.subset.size()[0:ndims]) == {1}
-            reduce_axes = [ndims]
+        assert input_expr.local_offset is not None
+        connectivity = dace_utils.connectivity_identifier(input_expr.local_offset)
+        connectivity_node = self.state.add_access(connectivity)
+        connectivity_desc = connectivity_node.desc(self.sdfg)
+        connectivity_desc.transient = False
+
+        desc = input_expr.dc_node.desc(self.sdfg)
+        if isinstance(input_expr, MemletExpr):
+            local_dim_indices = [i for i, size in enumerate(input_expr.subset.size()) if size != 1]
         else:
-            reduce_axes = None
+            local_dim_indices = list(range(len(desc.shape)))
+
+        assert len(local_dim_indices) == 1
+        local_dim_index = local_dim_indices[0]
+        assert desc.shape[local_dim_index] == offset_provider.max_neighbors
+
+        # we lower the reduction map with WCR out memlet in a nested SDFG
+        nsdfg = dace.SDFG("reduce_with_skip_values")
+        nsdfg.add_array(
+            "values",
+            (desc.shape[local_dim_index],),
+            desc.dtype,
+            strides=(desc.strides[local_dim_index],),
+        )
+        nsdfg.add_array(
+            "neighbor_indices",
+            (connectivity_desc.shape[1],),
+            connectivity_desc.dtype,
+            strides=(connectivity_desc.strides[1],),
+        )
+        nsdfg.add_scalar("acc", desc.dtype)
+        st_init = nsdfg.add_state("init")
+        st_init.add_edge(
+            st_init.add_tasklet(
+                "init_acc",
+                {},
+                {"__val"},
+                f"__val = {reduce_init.dc_dtype}({reduce_init.value})",
+            ),
+            "__val",
+            st_init.add_access("acc"),
+            None,
+            dace.Memlet(data="acc", subset="0"),
+        )
+        st_reduce = nsdfg.add_state_after(st_init, "reduce")
+        st_reduce.add_mapped_tasklet(
+            name="reduce_with_skip_values",
+            map_ranges=dict(i=f"0:{offset_provider.max_neighbors}"),
+            inputs={
+                "__val": dace.Memlet(data="values", subset="i"),
+                "__neighbor_idx": dace.Memlet(data="neighbor_indices", subset="i"),
+            },
+            code=f"__out = __val if __neighbor_idx != {gtx_common._DEFAULT_SKIP_VALUE} else {reduce_identity.dc_dtype}({reduce_identity.value})",
+            outputs={
+                "__out": dace.Memlet(data="acc", subset="0", wcr=reduce_wcr),
+            },
+            external_edges=True,
+        )
 
-        reduce_wcr = "lambda x, y: " + gtir_python_codegen.format_builtin(op_name, "x", "y")
-        reduce_node = self.state.add_reduce(reduce_wcr, reduce_axes, reduce_init.value)
+        nsdfg_node = self.state.add_nested_sdfg(
```

Comment:
What is about the strides that you use, I would say you have to pass them through the symbol mapping.

Reply:
The strides are exactly the same as the parent SDFG, so they are automatically mapped.

Reply:
Are you sure. I have never seen that feature.

Reply:
For both shape and strides, I allocate the inner arrays using the same symbols as in parent data descriptors:
```
        nsdfg.add_array(
            "values",
            (desc.shape[local_dim_index],),
            desc.dtype,
            strides=(desc.strides[local_dim_index],),
        )
        nsdfg.add_array(
            "neighbor_indices",
            (connectivity_desc.shape[1],),
            connectivity_desc.dtype,
            strides=(connectivity_desc.strides[1],),
        )
```
The symbols that are not present in the symbol mapping are automatically inherited from the parent SDFG. I know that if they were missing there, `add_nested_sdfg` would throw an error because of undefined symbol.